### ISSUE TITLE
Support 'require-sri-for' in Content-Security-Policy

### ISSUE THREAD

**Issue by RehanSaeed** [NWebsec/NWebsec#89]
I caught [this](https://twitter.com/Scott_Helme/status/792829254849486848) tweet from Scott Helme about the `require-sri-for` directive in CSP.

I already wrote an SRI `TagHelper` for ASP.NET Core which you can take a look at in my [blog post](http://rehansaeed.com/subresource-integrity-taghelper-using-asp-net-core/). It would be nice to be able to use this directive for an added layer of security.

**Comment by klings** [comment]
Yup, I see it's currently supported by Chrome, and Firefox will follow in march next year: https://bugzilla.mozilla.org/show_bug.cgi?id=1299483

I usually wait until there's support from more than one browser, but given the nature of this directive I'll squeeze it out sooner.

**Comment by RehanSaeed** [comment]
Cool 👍

**Comment by klings** [comment]
This is weird, `require-sri-for` is mentioned here: https://w3c.github.io/webappsec-subresource-integrity/
but I can't find any trace of it in the final spec: https://www.w3.org/TR/SRI/ nor can I find any traces of a webappsec discussion on dropping it.

Do you know what happened to it? It seems they dropped the idea to combine it with CSP altogether.

**Comment by klings** [comment]
It's also behind a flag in Chrome, https://www.chromestatus.com/feature/5635811978510336 and thus not widely available.

**Comment by RehanSaeed** [comment]
Not sure, perhaps @ScottHelme knows?

**Comment by ScottHelme** [comment]
Is it still behind a flag in Chrome? I added support to @reporturi and I get reports for the directive: https://scotthelme.co.uk/enforcing-the-use-of-sri/ 

I also can't find any mention of it on the flags page.

**Comment by RehanSaeed** [comment]
CSP 3.0 also adds support for [SRI hashes](https://chloe.re/2017/04/12/csp-hash-on-scripts-sri-and-require-sri-for-no-more-self/).

**Comment by klings** [comment]
This is still behind a flag in Chrome and FF.
